From d5b2aac92903ec0a573990bc94288196ea286697 Mon Sep 17 00:00:00 2001
From: Amber Lin <Amber.Lin@amd.com>
Date: Fri, 12 May 2017 10:56:39 -0400
Subject: [PATCH 2038/2094] drm/amdgpu: Disable SDMA-ULV interrupts

This is a temporary work around for intermittent SDMA hang seen on Vega10
when SDMA ULV interrupts are enabled. Revert this change once the fix in
the firmware is available.

Bug: SWDEV-121195

Change-Id: I4e58e4ce4828d95717a1e153d96ec05a5399c26d
Signed-off-by: Amber Lin <Amber.Lin@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/sdma_v4_0.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/sdma_v4_0.c b/drivers/gpu/drm/amd/amdgpu/sdma_v4_0.c
index 91cf7e6..07cb612 100644
--- a/drivers/gpu/drm/amd/amdgpu/sdma_v4_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/sdma_v4_0.c
@@ -669,6 +669,13 @@ static int sdma_v4_0_gfx_resume(struct amdgpu_device *adev)
 
 		if (adev->mman.buffer_funcs_ring == ring)
 			amdgpu_ttm_set_active_vram_size(adev, adev->mc.real_vram_size);
+
+		/* FIXME: temporarily disable SDMA-ULV interrupts for Vega10.
+		 * Remove this once the fix is in firmware.
+		 */
+		if (ring->adev->asic_type == CHIP_VEGA10)
+			WREG32(sdma_v4_0_get_reg_offset(i, mmSDMA0_ULV_CNTL),
+							0);
 	}
 
 	return 0;
-- 
2.7.4

